File: src/app/model/player.ts
import {Card} from './card';
import {ActType} from './actType';

export class Player {
  id: number;
  userId: string;
  firstCard: Card;
  secondCard: Card;
  lastAct: ActType;
  inRound: boolean;
  inRoom: boolean;
  chipCount: number;
  handType: string;
  seatNumber: number;
  access_token: string;

  static create() {
    return {
      id: 0,
      userId: '0',
      firstCard: undefined,
      secondCard: undefined,
      lastAct: undefined,
      inRound: true,
      inRoom: true,
      chipCount: 5000,
      handType: 'BAD',
      seatNumber: 1,
      access_token: 'ojfzeoio'
    };
  }
}

File: app/src/main/java/be/kdg/mobile_client/room/model/Player.java
package be.kdg.mobile_client.room.model;

import java.util.Objects;

import lombok.Data;

@Data
public class Player {
    private int id;
    private String userId;
    private String username;
    private Card firstCard;
    private Card secondCard;
    private ActType lastAct;
    private boolean inRound;
    private boolean inRoom;
    private int chipCount;
    private String handType;
    private int seatNumber;
    private String access_token;

    private transient boolean myTurn;
    private transient String bet;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Player player = (Player) o;
        return Objects.equals(userId, player.userId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(userId);
    }
}
